scripts/aggregate-translations/write-i18n-loaders.js

```diff
@@ -2,25 +2,50 @@ const path = require('path');
 const startCase = require('lodash.startcase');
 const supportedLocales = require('./i18nSupportedLocales');
 
-const createIntlLoader = (loaderName, locale) => (
-  `var ${loaderName} = function ${loaderName}() {
+const createIntlLoader = (loaderName, locale, format) => {
+  if (format === 'modern') {
+    return (
+      `const ${loaderName} = () =>
+   import('intl/locale-data/jsonp/${locale}.js')
+   .catch(error => console.log('An error occurred while loading intl/locale-data/jsonp/${locale}.js'));\n
+`
+    );
+  }
+
+  return (
+    `var ${loaderName} = function ${loaderName}() {
   return require.ensure([], function (require) {
     return require('intl/locale-data/jsonp/${locale}.js');
   }, '${locale}-intl-local');
 };\n
-`);
+`
+  );
+};
 
-const createTranslationLoader = (loaderName, locale) => (
-  `var ${loaderName} = function ${loaderName}(callback, scope) {
+const createTranslationLoader = (loaderName, locale, format) => {
+  if (format === 'modern') {
+    return (
+      `const ${loaderName} = (callback, scope) =>
+   import( /* webpackChunkName: "${locale}-translations" */ '${locale}.js')
```

Comment:
Does this syntax require Jest configuration changes since this will use wepackChunkName?

Reply:
Shouldn't. Configuring the .babelrc should be enough for this syntax to work.

Reply:
Is this something we should be providing documentation on or is this something we should expect consumers to understand how to setup and differentiate?

Reply:
We can do a documentation update to talk about this point.